routing/DELEGATED_CONTENT_ROUTING_HTTP.md

```diff
@@ -0,0 +1,184 @@
+
+The information required to write a record to a router (*"write" provider records*) may be different than the information contained when reading provider records (*"read" provider records*).
+
+For example, indexers may require a signature in `bitswap` write records for authentication of the peer contained in the record, but the read records may not include this authentication information.
+
+Both read and write provider records have a minimal required schema as follows:
+
+```json
+{
+    "Protocol": "<transfer_protocol_name>",
+    "Schema": "<transfer_protocol_schema>",
+    ...
+}
+```
+
+Where:
+
+- `Protocol` is the multicodec name of the transfer protocol
+- `Schema` denotes the schema to use for encoding/decoding the record
+  - This is separate from the `Protocol` to allow this HTTP API to evolve independently of the transfer protocol
+  - Implementations should switch on this when parsing records, not on `Protocol`
+- `...` denotes opaque JSON, which may contain information specific to the transfer protocol
+
+Specifications for some transfer protocols are provided in the "Transfer Protocols" section.
+
+## API
+
+### `GET /routing/v1/providers/{CID}`
+
+- Response codes
+  - `200`: the response body contains 0 or more records
+  - `404`: must be returned if no matching records are found
+  - `422`: request does not conform to schema or semantic constraints
+- Response Body
+  
+  ```json
+  {
+      "Providers": [
+          {
+              "Protocol": "<protocol_name>",
+              "Schema": "<schema>",
+              ...
+          }
+      ]
+  }
+  ```
+  
+- Default limit: 100 providers
+- Optional query parameters
+  - `transfer` only return providers who support the passed transfer protocols, expressed as a comma-separated list of transfer protocol names such as `transfer=bitswap,filecoin-graphsync-v1`
+  - `transport` for provider records with multiaddrs, only return records with multiaddrs explicitly supporting the passed transport protocols, encoded as decimal multicodec codes such as `transport=460,478` (`/quic` and `/tls/ws` respectively)
+- Implements pagination according to the Pagination section
+
+Each object in the `Providers` list is a *read provider record*.
+
+## Pagination
+
+APIs that return collections of results should support pagination as follows:
+
+- If there are more results, then a `NextPageToken` field should include an opaque string value, otherwise it should be undefined
+- The value of `NextPageToken` can be specified as the value of a `pageToken` query parameter to fetch the next page
+  - Character set is restricted to the regular expression `[a-zA-Z0-9-_.~]+`, since this is intended to be used in URLs
+- The client continues this process until `NextPageToken` is undefined or doesn't care to continue
+- A `pageLimit` query parameter specifies the maximum size of a single page
```

Comment:
nit: could this be renamed to `limit` ?  we dont have capital letters in other parameters